lib/spack/llnl/util/filesystem.py

```diff
@@ -2281,7 +2281,7 @@ def add_rpath(self, *paths):
     def _link(self, path, dest):
```

Comment:
A few requests:

* rename the `dest` parameter to `dest_dir`
* add a docstring mentioning that this skips cases where `path == dest_file` (to avoid errors when the user does not have developer mode enabled on Windows)
* (since this function is private to the class this is not essential but) it might be worth adding an explicit check that `dest_dir` is in fact a directory